core/src/main/scala/org/apache/spark/sql/delta/DeltaColumnMapping.scala

```diff
@@ -522,15 +522,17 @@ trait DeltaColumnMappingBase extends DeltaLogging {
   def isColumnMappingReadCompatible(newMetadata: Metadata, oldMetadata: Metadata): Boolean = {
     val (oldMode, newMode) = (oldMetadata.columnMappingMode, newMetadata.columnMappingMode)
     if (oldMode != NoMapping && newMode != NoMapping) {
+      require(oldMode == newMode, "changing mode is not supported")
```

Comment:
do you have a test for this?

Reply:
This is just defense in depth (or me being paranoid :)), we should've already blocked column mapping changing modes, e.g. [here](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/DeltaColumnMapping.scala#L119)